### PR TITLE
fix: decode substrate multilocation

### PR DESCRIPTION
**Pull request by tcar121293**
decode substrate multilocation
## Description
decode substrate multilocation and store actual substrate address
## Related Issue Or Context

Closes: #76 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: src/indexer/utils/evm/index.ts [hunk 1]
```diff
@@ -91,35 +101,40 @@ export async function getDecodedLogs(
...
-): DecodedDepositLog {
+): Promise<DecodedDepositLog> {
   const resourceType = resourceMap.get(decodedLog.args.resourceID as string)?.type || ""
   const resourceDecimals = resourceMap.get(decodedLog.args.resourceID as string)?.decimals || 18
-
   const arrayifyData = getBytes(decodedLog.args.data as BytesLike)
-  const filtered = arrayifyData.filter((_, idx) => idx + 1 > 65)
-  const hexAddress = hexlify(filtered)
+
+  let destination = ""
```
Comment (mpetrun5) [inline src/indexer/utils/evm/index.ts:124]: Move to a separate function.
Would be nice if slowly tart testing stuff so also test that function.

File: src/indexer/utils/evm/index.ts [hunk 2]
```diff
@@ -130,6 +145,35 @@ export function parseDeposit(
   }
 }
 
+function parseEvmDestination(bytes: Uint8Array): string {
+  const filtered = bytes.filter((_, idx) => idx + 1 > 65)
+  return hexlify(filtered)
+}
+
+async function ParseSubstrateDestination(bytes: Uint8Array, domain: Domain): Promise<string> {
```
Comment (mpetrun5) [inline src/indexer/utils/evm/index.ts:153]:
```suggestion
async function parseSubstrateDestination(bytes: Uint8Array, domain: Domain): Promise<string> {
```

File: src/indexer/utils/evm/index.ts [hunk 3]
```diff
@@ -130,6 +145,35 @@ export function parseDeposit(
...
+  })
+
+  const recipientlen = Number("0x" + Buffer.from(bytes.slice(32, 64)).toString("hex"))
+
+  const recipient = "0x" + Buffer.from(bytes.slice(64, 64 + recipientlen)).toString("hex")
+
+  const decodedData = api.createType("MultiLocation", recipient)
+  const multiAddress = decodedData.toJSON() as unknown as MultiLocation
+
+  for (const [, junctions] of Object.entries(multiAddress.interior)) {
```
Comment (mpetrun5) [inline src/indexer/utils/evm/index.ts:167]: Isn't there a function in polkadot.js lib to do this?

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: src/indexer/utils/evm/index.ts (hunk 1 above)
Comment (mpetrun5) [reply src/indexer/utils/evm/index.ts:124]: (if possible parseDeposit as well)

File: src/indexer/utils/evm/index.ts (hunk 3 above)
Comment (tcar121293) [reply src/indexer/utils/evm/index.ts:168]: I think not, at least i didn't find it

**Review by mpetrun5 — APPROVED** [review]

**Comment by mpetrun5** [comment]
Fix lint and e2e tests and please also add the test action to `Lint, Build, Test` action.

**Pull request merged**